aten/src/ATen/BatchingUtils.h

```diff
@@ -0,0 +1,42 @@
+#include <ATen/BatchedTensorImpl.h>
+
+namespace at {
+
+/*
+ * Utility functions used to implement batching rules.
+ */ 
+
+// Creates a bitset of all the levels present in `bdims`.
+//
+// For example:
+//   createLevelsBitset([(lvl=1, dim=2), (lvl=3, dim=1)]) -> 1010000...
+std::bitset<kVmapNumLevels> createLevelsBitset(BatchDimsRef bdims);
+
+// Produces a new vector of the same size as `arr` where each element of is
+// created by calling `func` on  each element of `arr` in order.
+// Equivalent to `output = map(func, arr)` in Python.
+std::vector<int64_t> transform(IntArrayRef arr, std::function<int64_t(int64_t)> func);
```

Comment:
This is a pretty generic name that's been put in the top level `at` namespace. Try making it more descriptive or putting it in a subnamespace.

Reply:
Good point. I made it more descriptive. In the long term I want to move most of the BatchedTensor things into its own subnamespace to avoid crowding the at:: namespace